core/src/identity.rs

```diff
@@ -251,23 +254,19 @@ impl PublicKey {
     /// Encode the public key into a protobuf structure for storage or
     /// exchange with other nodes.
     pub fn to_protobuf_encoding(&self) -> Vec<u8> {
-        use prost::Message;
+        use protobuf::Message;
 
         let public_key = keys_proto::PublicKey::from(self);
 
-        let mut buf = Vec::with_capacity(public_key.encoded_len());
-        public_key
-            .encode(&mut buf)
-            .expect("Vec<u8> provides capacity as needed");
-        buf
+        public_key.write_to_bytes().expect("Encoding failed.")
```

Comment:
Can we say something like "all fields to be initialized" given that's the invariant we are relying on?